GoogleDataTransport/GDTCORLibrary/GDTCORPlatform.m

```diff
@@ -249,35 +255,32 @@ - (BOOL)isAppExtension {
 #endif
 }
 
-/** Returns a UIApplication or WKExtension instance if on the appropriate platform.
- *
- * @return The shared UIApplication or WKExtension if on the appropriate platform.
- */
 #if TARGET_OS_IOS || TARGET_OS_TV
 - (nullable UIApplication *)sharedApplicationForBackgroundTask {
 #elif TARGET_OS_WATCH
-- (nullable WKExtension *)sharedApplicationForBackgroundTask {
+- (nullable NSProcessInfo *)sharedNSProcessInfoForBackgroundTask {
 #else
 - (nullable id)sharedApplicationForBackgroundTask {
 #endif
   if ([self isAppExtension]) {
     return nil;
   }
-  id sharedApplication = nil;
+  id sharedInstance = nil;
 #if TARGET_OS_IOS || TARGET_OS_TV
   Class uiApplicationClass = NSClassFromString(@"UIApplication");
   if (uiApplicationClass &&
       [uiApplicationClass respondsToSelector:(NSSelectorFromString(@"sharedApplication"))]) {
-    sharedApplication = [uiApplicationClass sharedApplication];
+    sharedInstance = [uiApplicationClass sharedApplication];
   }
 #elif TARGET_OS_WATCH
-  Class wkExtensionClass = NSClassFromString(@"WKExtension");
-  if (wkExtensionClass &&
-      [wkExtensionClass respondsToSelector:(NSSelectorFromString(@"sharedExtension"))]) {
-    sharedApplication = [wkExtensionClass sharedExtension];
+  // The processInfo class method returns the shared agent for the current process.
+  Class nsProcessInfoClass = NSClassFromString(@"NSProcessInfo");
+  if (nsProcessInfoClass &&
+      [nsProcessInfoClass respondsToSelector:(NSSelectorFromString(@"processInfo"))]) {
+    sharedInstance = [nsProcessInfoClass processInfo];
   }
 #endif
```

Comment:
Just checking: are we going to take care of macOS here or it is supposed to be do nothing.

Reply:
This sharedInstance are taking responsibility for handling some task to be continually running when app goes into background, I think in macOS, this situation would not be considered. I am not sure about that. I would consult Michael as well.

Reply:
I'm asking because it seems like you are handling macOS at line 265.

Reply:
Seems there are some waste checks for macOS here, I will carefully consider that. Thank you.

Reply:
Just for clarity: macOS doesn't really have an app lifecycle analogous to anything we do on any other platforms. For our purposes: macOS doesn't have a background state, just a started and stopping state.